sql/cell_value.cc

```diff
@@ -199,14 +199,22 @@ CellValue
 CellValue::
 parse(const char * s_, size_t len, StringCharacteristics characteristics)
 {
-    std::string cstring(s_, len); // ensure the buffer is null terminated
-    const char * s = cstring.c_str();
+    static constexpr size_t NUMERICAL_BUFFER = 64;
+
+    // if the string is longer than 64 characters it can't realistically
+    // be a numerical value
+    if (len > NUMERICAL_BUFFER)
```

Comment:
I guess it's OK, though we should clearly document in the natural_cast documentation that numbers must have a length < 64, and put an issue (soon!) to fix it.  For example, 0.0000000000000000000000000000000000000000000000000000000000000000000000000000000023942190 is a valid number that could easily be produced by a bad format string.